src/code42cli/cmds/auditlogs.py

```diff
@@ -0,0 +1,193 @@
+
+EVENT_KEY = "events"
+AUDIT_LOGS_KEYWORD = "audit-logs"
+
+AUDIT_LOGS_DEFAULT_HEADER = OrderedDict()
+AUDIT_LOGS_DEFAULT_HEADER["timestamp"] = "Timestamp"
+AUDIT_LOGS_DEFAULT_HEADER["type"] = "Type"
+AUDIT_LOGS_DEFAULT_HEADER["actorName"] = "ActorName"
+AUDIT_LOGS_DEFAULT_HEADER["actorIpAddress"] = "ActorIpAddress"
+AUDIT_LOGS_DEFAULT_HEADER["success"] = "Success"
+AUDIT_LOGS_DEFAULT_HEADER["resultCount"] = "ResultCount"
+
+filter_option_usernames = click.option(
+    "--username", required=False, help="Filter results by usernames.", multiple=True,
+)
+filter_option_user_ids = click.option(
+    "--user-id", required=False, help="Filter results by user ids.", multiple=True,
+)
+
+filter_option_user_ip_addresses = click.option(
+    "--user-ip",
+    required=False,
+    help="Filter results by user ip addresses.",
+    multiple=True,
+)
+filter_option_affected_user_ids = click.option(
+    "--affected-user-id",
+    required=False,
+    help="Filter results by affected user ids.",
+    multiple=True,
+)
+filter_option_affected_usernames = click.option(
+    "--affected-username",
+    required=False,
+    help="Filter results by affected usernames.",
+    multiple=True,
+)
+filter_option_event_types = click.option(
+    "--event-type",
+    required=False,
+    help="Filter results by event types.",
+    multiple=True,
+)
+
+
+def filter_options(f):
+    f = begin_option(
+        f,
+        AUDIT_LOGS_KEYWORD,
+        callback=lambda ctx, param, arg: parse_min_timestamp(arg),
+        required=True,
+    )
+    f = end_option(
+        f, AUDIT_LOGS_KEYWORD, callback=lambda ctx, param, arg: parse_max_timestamp(arg)
+    )
+    f = filter_option_event_types(f)
+    f = filter_option_usernames(f)
+    f = filter_option_user_ids(f)
+    f = filter_option_user_ip_addresses(f)
+    f = filter_option_affected_user_ids(f)
+    f = filter_option_affected_usernames(f)
+    return f
+
+
+@click.group(cls=OrderedGroup)
+@sdk_options(hidden=True)
+def audit_logs(state):
+    """Retrieve audit logs."""
+    pass
+
+
+@audit_logs.command()
+@filter_options
+@format_option
+@sdk_options()
+def search(
+    state,
+    begin,
+    end,
+    event_type,
+    username,
+    user_id,
+    user_ip,
+    affected_user_id,
+    affected_username,
+    format,
+):
+    """Search audit logs."""
+    _search(
+        state.sdk,
+        format,
+        begin_time=begin,
+        end_time=end,
+        event_types=event_type,
+        usernames=username,
+        user_ids=user_id,
+        user_ip_addresses=user_ip,
+        affected_user_ids=affected_user_id,
+        affected_usernames=affected_username,
+    )
+
+
+@audit_logs.command()
+@filter_options
+@server_options
+@send_to_format_options
+@sdk_options()
+def send_to(
+    state,
+    hostname,
+    protocol,
+    format,
+    begin,
+    end,
+    event_type,
+    username,
+    user_id,
+    user_ip,
+    affected_user_id,
+    affected_username,
+):
+    """Send audit logs to given server address."""
```

Comment:
Can we say `to _the_ given server address` to be consistent with the security-data send-to command? And I also noticed when checking to see what the help text was on security-data that `saved-search` is missing help text:

```
Commands:
  clear-checkpoint  Remove the saved file event checkpoint from `--use-checkpoint/-c` mode.
  search            Search for file events.
  saved-search
  send-to           Send events to the given server address.
```

Could you update that too?

Reply:
Changed.